apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/content/staticelement/PageStaticDeletor.java

```diff
@@ -0,0 +1,51 @@
+package gov.cdc.nbs.questionbank.page.content.staticelement;
+
+import javax.persistence.EntityManager;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import gov.cdc.nbs.questionbank.entity.WaTemplate;
+import gov.cdc.nbs.questionbank.entity.WaUiMetadata;
+import gov.cdc.nbs.questionbank.entity.repository.WaUiMetadataRepository;
+import gov.cdc.nbs.questionbank.page.content.staticelement.exceptions.DeleteStaticElementException;
+import gov.cdc.nbs.questionbank.page.content.staticelement.request.DeleteElementRequest;
+
+@Component
+@Transactional
+public class PageStaticDeletor {
+    private final WaUiMetadataRepository uiMetadatumRepository;
+    private final EntityManager entityManager;
+
+    public PageStaticDeletor(
+            WaUiMetadataRepository uiMetadataRepository,
+            EntityManager entityManager) {
+        this.uiMetadatumRepository = uiMetadataRepository;
+        this.entityManager = entityManager;
+    }
+
+    public boolean deleteStaticElement(Long pageId, DeleteElementRequest request) {
+        if (pageId == null) {
+            throw new DeleteStaticElementException("Page is required");
+        }
+
+
+        WaTemplate template = entityManager.find(WaTemplate.class, pageId);
+
+        if (template == null) {
+            throw new DeleteStaticElementException("Please provide valid page ID");
+        }
+
+        if (!template.getTemplateType().equals("Draft")) {
```

Comment:
nit (non-blocking): its good to do the comparison in the reverse order just in case the templateType is null.
`"Draft".equals(template.getTemplateType());`

Reply:
Can pages only have a type of "Drafts"?